kong/plugins/tcp-log/handler.lua

```diff
@@ -43,6 +43,29 @@ function TcpLogHandler:new()
   TcpLogHandler.super.new(self, "tcp-log")
 end
 
+function TcpLogHandler:access(conf)
+  TcpLogHandler.super.access(self)  
+
+  if conf.log_body then
+    ngx.req.read_body()
+    ngx.ctx.req_resp_bodies = {
+      request_body = ngx.req.get_body_data(),
```

Comment:
This method call will fail with requests with long bodies. For those, nginx uses temporary fields instead. Here's an example of how to read the body data: https://github.com/Kong/kong/blob/master/spec/fixtures/mock_upstream.lua#L165-L183